examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb

```diff
@@ -0,0 +1,512 @@
+    "    samples.append(labels)\n",
+    "    sentence= [word for word in sentence if word !=\"<start>\" if word!=\"<end>\"]\n",
+    "    clean_sterbeliste.append(sentence)\n",
+    "\n",
+    "print(f\"Length of the sentence: {len(sentence)}\")\n",
+    "print(f\"Sample_tagging:{samples[0]}\")\n",
+    "print(f\"Number of place names: {len(locations)}\")\n",
+    "print(f\"=======================================\")\n",
+    "assert len(sentence) == len(labels)\n",
+    "#print(f\"Sentence and labeles: {[(word, label) for (word, label) in zip(sentence, labels)]}\")\n",
+    "print(f\"Cleaned Sentence: {clean_sterbeliste[0]}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## 3.Building training data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 118,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "'''\n",
+    "X Matrix\n",
+    "\n",
+    "Dimensions # sentences x # sentences (one column samples df, len(sample))\n",
+    "\n",
+    "'''\n",
+    "X = pd.DataFrame(clean_sterbeliste)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "**T Matrix**\n",
+    "\n",
+    "np.sparse matirx 1 colum 0, 1 colum 1, rows are keywords\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 117,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(6, 2)"
+      ]
+     },
+     "execution_count": 117,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "'''\n",
+    "T Matrix\n",
+    "Dimensions: # place names x # 2\n",
+    "\n",
+    "'''\n",
+    "import numpy as np\n",
+    "a= np.zeros(len(locations))\n",
+    "b= np.full(len(locations),1)\n",
+    "\n",
+    "T = np.stack((b,a.T))\n",
+    "T.T.shape\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Z Matrices**\n",
+    "Each sentense is going to be its own Z Matrix.\n",
+    "colums = place names\n",
+    "rows = words in sentense"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 119,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "'''\n",
+    "Z Matrices\n",
+    "\n",
+    "In the next function, we are building the Z Matrixes by looping through all samples.\n",
+    "Each sample is compared with the list of locations, if the location is in the sample,\n",
+    "the new list \"matched_loc\" gets an entry 1 otherwise a 0 is added\n",
+    "Then the list is transformed to an numpy array and brought into shape:\n",
+    "\n",
+    "Dimensions\n",
+    "number of place names x number of words\n",
+    "\n",
+    "'''\n",
+    "locations_list = list(range(0,(len(locations))))\n",
+    "\n",
+    "\n",
+    "for sample in samples:\n",
```

Comment:
again: it is better to convert this part of code into the separate function + write a test for it (let's talk about it)